test: fix unstable `test_detect_port_with_cache`

Sometimes the test is unstable. For example, https://github.com/espressif/pytest-embedded/actions/runs/6783180237/job/18436888658

Fails for unrelated changes.

Hello, @hfudev!

Do you mean this type of error? Or is there something else that I need to notice?
**test**: test_teardown_called_for_multi_dut?
```
>           self.fd = os.open(self.portstr, os.O_RDWR | os.O_NOCTTY | os.O_NONBLOCK)
E           FileNotFoundError: [Errno 2] No such file or directory: '/dev/ttyUSB100'
```

--- a/conftest.py
+++ b/conftest.py
@@ -18,6 +18,14 @@ def copy_fixtures(testdir):
     yield
 
 
+@pytest.fixture(autouse=True)
+def cache_file_remove(cache_dir):
+    yield
+    _cache_file_path = os.path.join(cache_dir, 'port_target_cache')
+    if os.path.exists(_cache_file_path):
+        os.remove(_cache_file_path)
+
+
 @pytest.fixture
 def first_index_of_messages():
     def _fake(_pattern: Pattern, _messages: List[str], _start: int = 0) -> int:

